cups/tls-darwin.c

```diff
@@ -441,16 +444,15 @@ httpCopyCredentials(
 
   if (!(error = SSLCopyPeerTrust(http->tls, &peerTrust)) && peerTrust)
   {
+    DEBUG_printf(("2httpCopyCredentials: Peer provided %d certificates.", (int)SecTrustGetCertificateCount(peerTrust)));
+
     if ((*credentials = cupsArrayNew(NULL, NULL)) != NULL)
     {
-      CFArrayRef secArray = SecTrustCopyCertificateChain(peerTrust);
-      CFIndex i, count = CFArrayGetCount(secArray);
+      count = SecTrustGetCertificateCount(peerTrust);
```

Comment:
This should be the only file reverted. 

Every other change builds on all supported OS versions.